### ISSUE TITLE
I.L. FE: Review page should not be using table markup

### ISSUE THREAD

**Issue by laflannery** [department-of-veterans-affairs/va.gov-cms#14047]
## Description
The review information screen on the Income Limits app is currently presented using a table, however this is done for layout only - meaning this table is not needed for data analysis reasons but just to lay  the information out in a nice, column way. We should only use this as a very last resort, presentation tables should really never be the answer. It makes reading and understanding the data more complex than it needs to be for screen reader users.

### User story

**AS A** Veteran using a screen reader
**I WANT** information presented using simple semantic HTML
**SO THAT** I can review my information as easily as possible.

### Engineering notes / background
I would like to use an unordered list and then flexbox to position it. I did this locally, just to make sure it was reading correctly so this is a barebones, basic markup of what this might look like:
![image](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/106678594/8c484ddd-c75d-433e-b7e2-4e2c9ebae844)

**Main things to keep in mind:**
- There was an `aria-hidden` attribute on the <br> before each value. We want to remove this or the values, the info you are trying to review will not be read by screen readers
- By wrapping this all in a list, the user will immediately know how many items they have to review
- The "Edit" should be at the end of the list item. This would be the logical placement as they would review each item and then choose whether or not to go edit or not.

**Question on Edit link:**
- Does the "Edit" need to be a `<button>`? For material honesty, this should actually be a link because this is navigating to a new page but I'm unclear on the functional app requirements here and if a `<button>` element is required.
- HOWEVER, I'm looking at the <va-link> component and I do not see an available aria-label property so I don't think if you used it, you could put the correct, unique labels like you did on the existing buttons.
- If that is the case, and you can't label them correctly, we should NOT use the link component.

## Acceptance criteria

- [ ] All information the user is expected to review is wrapped in an unordered list
- [ ] No `aria-hidden` attributes are used
- [ ] Order of each list item is Label, value, Edit link
- [ ] Edit action is a link if possible

**Comment by randimays** [comment]
@laflannery The `Edit` button cannot be a link for one main reason:
Our app is not going to support deep-linking, or navigating directly to `/zip`, for example. If any of our links try to do that, it will simply direct back to the home page instead. The behavior isn't currently this way in staging, but I am working on a ticket that will accomplish this.

**Comment by randimays** [comment]
@laflannery Actually, rethinking this a little. That other ticket I mentioned that is in progress is going to take some time to complete as I have to wait for a Wednesday deploy. After that is finished, I will check if we can convert to a link successfully with some logic to distinguish between "edit mode" and regular flow. If it's ok with you and @wesrowe, I can add an AC to [this ticket](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/13593) since it all has to do with routing.

**Comment by wesrowe** [comment]
@randimays, confirming – did you mean to close this ticket? I was just a little surprised that the comments ended with your thoughts and not a response/approval from @laflannery. Also, the comments were about the Edit links/buttons, not the table markup in the ticket title. 

But I didn't quite follow all the cross-ticket complications above, so I could very easily be missing something.

**Comment by randimays** [comment]
@wesrowe Ah, I forgot about the above comment. My apologies. I got Laura's approval on the PR and moved it through without remembering this.

@wesrowe @laflannery Should we add an AC to [this ticket](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/13593) since it's all routing related? If not, we can reopen this one.

**Comment by wesrowe** [comment]
I'm fine with adding an AC to the other ticket – what would that AC be?

**Comment by randimays** [comment]
I think @laflannery's proposal was good:
``` Edit action is a link if possible```
